### PR TITLE
Fixed a typo and reordered items.

### PR DESCRIPTION
**Pull request by dalinaum**
A simple fix again. Please review this @realm/java

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 0.84.2
- * Fixed a bug making it impossible to read Realms created by iOS if a class contained primary keys (#1703).
  * Added Realm.isEmpty().
- * Fixed bug making it impossible to convert a field to being required during a migration.
+ * Fixed a bug making it impossible to convert a field to being required during a migration.
```
Comment (zaki50) [inline changelog.txt]: I think it's better to add PR number(#1695).

**Comment by zaki50** [comment]
:+1: other than my comment.

**Comment by zaki50** [comment]
:+1:

File: changelog.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 0.84.2
- * Fixed a bug making it impossible to read Realms created by iOS if a class contained primary keys (#1703).
  * Added Realm.isEmpty().
- * Fixed bug making it impossible to convert a field to being required during a migration.
+ * Fixed a bug making it impossible to convert a field to being required during a migration (#1695).
```
Comment (nhachicha) [inline changelog.txt]: to being -> to become ?

File: changelog.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
+ * Fixed a bug making it impossible to read Realms created by iOS if a class contained primary keys (#1703).
```
Comment (nhachicha) [inline changelog.txt]: Fixed a bug making it impossible to read Realms using primary keys and created from iOS

File: changelog.txt [hunk 4]
```diff
@@ -9,7 +9,7 @@
  * Updated ProGuard configuration. See documentation https://realm.io/docs/java/latest/#proguard for more details.
  * Updated Kotlin example to use 1.0.0-beta.
  * Fixed warnings reported by `lint -Xlint:all` (#1644).
- * Fixed a bug where simultaneous opening and closing a Realm on diffesrent threads might result in a NullPointerException (#1646).
+ * Fixed a bug where simultaneous opening and closing a Realm on different threads might result in a NullPointerException (#1646).
```
Comment (nhachicha) [inline changelog.txt]: opening and closing a Realm -> opening and closing of a Realm from different ...

**Comment by dalinaum** [comment]
Thanks @nhachicha. I modified them.

**Comment by cmelchior** [comment]
:+1:

**Pull request merged**